labs/a9.md

```diff
@@ -45,39 +66,740 @@ Allowing clients to install their own security protects allows them protection
+substantially if they are able to influence the direction of the project in
+their favor. You can avoid this by signing your messages: in the beginning, you
+would publish Natoshi's public key, and thereafter, for every post you make, you
+would encrypt the content of the message using your (Natoshi's) private key and
+post that along with your original message. Then, anyone who wants to verify
+that Natoshi (i.e. the owner of the private key corresponding to the public key
+that belongs to Natoshi) did in fact publish a particular message can simply
+use Natoshi's public key to decrypt the encrypted signature and compare the
+content against the original message. Pretenders to Natoshi's throne will be
+unable to sign their false statements such that they can be verified with
+Natoshi's published public key because they don't have Natoshi's private key,
+and you can rest assured that no one will unduly influence your project in your
+name while you go into hiding from the IRS and DEA, unless they happen to have
+warehouses full of ASICs and lots of cheap electricity.
+
+**However, in this scheme, how do you prevent an adversary from publishing a fake
+public key and claiming to be you** (they can make valid signatures against that
+fake public key)? Somehow, you need to "bootstrap" trust: someone would need to
+verify your identity and publicly affirm that your public key actually
+corresponds to you. We do this by means of **certificates**: a signed statement
+claiming that a particular public key does, in fact, belong to who it claims to
+belong to. Who signs this certificate? A **certificate authority**, someone we
+trust to be responsible about verifying identities and issuing signatures. But
+how do we know which CAs to trust, and how can we trust that a CA that claims
+to be trustworthy actually is? They probably need a certificate as well. It
+sounds like it might be turtles all the way down; however, the chain does end
+somewhere: the so-called root of trust, the root CAs. These are the CAs whose
+certificates are pre-installed by browsers and operating systems and therefore
+intrinsically trusted, without any further certificates necessary. If a root CA
+signs your certificate, we assume they've done the due diligence necessary to
+be willing to risk their reputation by signing your certificate, and basically
+take their word for it. This model, known as the "Web of Trust" is how network
+security works today. Unfortunately, it isn't as reliable as we may have hoped:
+some CAs are scummy and will sign anything for enough money, resulting in valid
+certificates being issued for domains like microsoft.com and github.com to
+entities who are very obviously not Microsoft or GitHub.[^badwosign] Furthermore,
+any entity with enough border control can force the installation of their own
+root certificates (e.g. the government of Kazakhstan[^badkazakh]) and intercept
+any traffic by issuing their own bogus certificates for any domain.
```

Comment:
This could also be broken up with some bullet points and made more readable. Take advantage of all of markdowns features